### PR TITLE
Naming conventions:private members and parameters.

### PR DESCRIPTION
**Pull request by TotoGaz**
Hi @joshua-white I've been working on naming conventions.

I've been running `clang-tidy` (see [website](https://clang.llvm.org/extra/clang-tidy/)) on GEOSX.
I did not consider the submodules because crossed PR are more complicated.

Please have a look at the `.clang-tidy` file that mainly defines the (first) fixes that I have asked `clang-tidy` to apply.
Have a look at [all the options too](https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html)!

Not everything is automatically perfect and I had to correct some mistakes sometimes (mostly with `lambdas`).
Strong change is `dPhaseViscosity_dPressure` to `dPhaseViscosityDPressure `  _e.g._.
Avoiding underscores may be questionnable sometimes too.
But uniformity may be useful (no automatic validation without this, _e.g._), so...

This is a first shot, we can define what we want and I run the script and corrections again on a (new ?) PR.
`conflicts` need to be minimized.

Please check the result to see whether we want to go that way.

I'll probably need to `uncrustify`.

PS: Surely needs to work on the `doxygen` parameters that are not considered by `clang-tidy`. Requires time, but i will be OK.

### CODE REVIEW

**Comment by joshua-white** [comment]
This is a great strategy, and should save a lot of work.  My comments:

- I noticed you focused only on the private members.  I personally think the public methods are the most important to make constant across the code base.  Internal implementations can be more flexible in their conventions.  What you have looks pretty great though, except ...
- On `dxDy`, I suspect that will cause a lot of controversy.  I am okay with it, but it would take some getting used to.

**Comment by TotoGaz** [comment]
> * I noticed you focused only on the private members.

I did focus on private members first in order to earn some experience with this kind of tool in a simpler situation.
For example, some manual actions still remain (actually more than I expected).

Public information should follow. It could be complicated to do everything in the same time.

I would like to be able to run the script, perform my checks and push within a couple of hours to limit conflicts.

> * On `dxDy`, I suspect that will cause a lot of controversy.  I am okay with it, but it would take some getting used to.

This is also open for discussion. There may be some cases where we prefer `dx_dy`: up to the team to decide.
But if we are not homogeneous, it will be more complicated to impose automatic testing I think (_e.g._).

File: src/coreComponents/codingUtilities/Utilities.hpp [hunk 1]
```diff
@@ -144,32 +144,32 @@ void copy( localIndex const N, VEC1 const & v1, VEC2 const & v2 )
 template< typename MATRIX, typename VEC1, typename VEC2 >
 GEOSX_HOST_DEVICE
 GEOSX_FORCE_INLINE
-void applyChainRule( localIndex const N,
-                     MATRIX const & dy_dx,
-                     VEC1 const & df_dy,
-                     VEC2 && df_dx )
+void applyChainRule( localIndex const n,
+                     MATRIX const & dyDx,
```
Comment (rrsettgast) [inline src/coreComponents/codingUtilities/Utilities.hpp:148]: This has to be kept the way it was. `camelCase` is for spoken word, but the `_` signifies some sort of operator or break in the phrase. In this case, it is `dy/dx` but we can't write it like that. We could do `dy_div_dx` but that is a little too verbose and really isn't what is happening anyways. So the use of `dy_dx` indicates that there is some sort of operator between the terms....leaving the reader to figure that out...which I think is reasonable.

My objection applies to every other replacement of derivative terms.

File: src/coreComponents/common/BufferAllocator.hpp [hunk 2]
```diff
@@ -64,7 +64,7 @@ class BufferAllocator
 private:
   // An umpire allocator allocating the type for which this class is instantiated.
   umpire::TypedAllocator< value_type > m_alloc;
-  bool m_prefer_pinned_l;
+  bool m_preferPinnedL;
```
Comment (rrsettgast) [inline src/coreComponents/common/BufferAllocator.hpp:67]: The `L` would need to be expanded here.

File: src/coreComponents/constitutive/ExponentialRelation.hpp [hunk 3]
```diff
@@ -72,10 +72,10 @@ struct ExponentialCompute< T, ExponentApproximationType::Full, false >
 
   GEOSX_HOST_DEVICE
   GEOSX_FORCE_INLINE
-  static void Compute( const T & x0, const T & y0, const T & alpha, const T & x, T & y, T & dy_dx )
+  static void Compute( const T & x0, const T & y0, const T & alpha, const T & x, T & y, T & dyDx )
```
Comment (rrsettgast) [inline src/coreComponents/constitutive/ExponentialRelation.hpp:75]: 👎

File: src/coreComponents/constitutive/capillaryPressure/BrooksCoreyCapillaryPressure.hpp [hunk 4]
```diff
@@ -39,11 +39,11 @@ class BrooksCoreyCapillaryPressureUpdate final : public CapillaryPressureBaseUpd
                                       arrayView1d< integer const > const & phaseTypes,
                                       arrayView1d< integer const > const & phaseOrder,
                                       arrayView3d< real64 > const & phaseCapPressure,
-                                      arrayView4d< real64 > const & dPhaseCapPressure_dPhaseVolFrac )
+                                      arrayView4d< real64 > const & dPhaseCapPressureDPhaseVolFrac )
```
Comment (rrsettgast) [inline src/coreComponents/constitutive/capillaryPressure/BrooksCoreyCapillaryPressure.hpp:42]: 👎

File: src/coreComponents/constitutive/fluid/CompositionalMultiphaseFluid.cpp [hunk 5]
```diff
@@ -38,13 +38,13 @@ namespace
 
 PVTPackage::EOS_TYPE getCompositionalEosType( string const & name )
 {
-  static std::map< string, PVTPackage::EOS_TYPE > const eosTypes =
+  static std::map< string, PVTPackage::EOS_TYPE > const EOS_TYPES =
```
Comment (rrsettgast) [inline src/coreComponents/constitutive/fluid/CompositionalMultiphaseFluid.cpp:41]: why ALL_CAPS?

File: src/coreComponents/constitutive/fluid/PVTFunctions/CO2SolubilityFunction.cpp [hunk 6]
```diff
@@ -42,56 +42,56 @@ constexpr real64 acoef[] =
  8.93353441e-5, 7.88998563e-5, -1.66727022e-2, 1.398, 2.96e-2};
 
 
-real64 ff( real64 const & T, real64 const & P, real64 const & V_r )
```
Comment (rrsettgast) [inline src/coreComponents/constitutive/fluid/PVTFunctions/CO2SolubilityFunction.cpp:45]: `T` is standard short for temperature?
`P` is standard short for pressure.
`V_r` is probably `volume_r`...don't know what r stands for.
@yue-2018 @francoishamon @klevzoff ?

at any rate, don't think this should be applied here.

File: src/coreComponents/constitutive/fluid/PVTFunctions/FenghourCO2ViscosityFunction.cpp [hunk 7]
```diff
@@ -120,7 +120,7 @@ void FenghourCO2ViscosityFunction::Evaluation( EvalVarArgs const & pressure, Eva
   value.m_der[0] = viscosity.m_der[0];
 }
 
-void FenghourCO2ViscosityFunction::FenghourCO2Viscosity( real64 const & Tcent, real64 const & den, real64 & vis )
+void FenghourCO2ViscosityFunction::FenghourCO2Viscosity( real64 const & tCent, real64 const & den, real64 & vis )
```
Comment (rrsettgast) [inline src/coreComponents/constitutive/fluid/PVTFunctions/FenghourCO2ViscosityFunction.cpp:123]: Temperature in Celsius. Not a fan of `tCent`

**Review by rrsettgast — CHANGES_REQUESTED** [review]

**Comment by joshua-white** [comment]
I think perfect consistency is a noble goal, but even small steps might help a lot.  Does it make sense to just focus on the public methods first?  It may be super hard to enforce consistency across private implementations and parameter naming, as there are a million reasons to break the rules. 

@TotoGaz I'm guessing we don't have much tuning flexibility beyond the options you showed in your script?

**Comment by TotoGaz** [comment]
> I think perfect consistency is a noble goal, but even small steps might help a lot. Does it make sense to just focus on the public methods first? It may be super hard to enforce consistency across private implementations and parameter naming, as there are a million reasons to break the rules.
> 
> @TotoGaz I'm guessing we don't have much tuning flexibility beyond the options you showed in your script?

The options basically are (see https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html)
```
lower_case
UPPER_CASE
camelBack
CamelCase
camel_Snake_Back
Camel_Snake_Case
aNy_CasE
```

OK to give a shot for public member functions, we will see what it brings.
I'm a bit worried about `traits` classes, I think I'll need to select carefully the options.

File: src/coreComponents/constitutive/fluid/CompositionalMultiphaseFluid.cpp (hunk 5 above)
Comment (TotoGaz) [reply src/coreComponents/constitutive/fluid/CompositionalMultiphaseFluid.cpp:41]: `static const` are often upper case. We can remove if needed.

**Comment by TotoGaz** [comment]
Closing until further work during next doc day.

**Pull request closed without merging**